chore(docs): updating Pokeshop documentation (#1892)
kubeshop/tracetest: docs/docs/live-examples/pokeshop/use-cases/import-pokemon.md

# Pokeshop API - Import Pokemon

This use case showcases a more complex scenario involving an async process. Usually, when working with microservices, there are use cases where some of the processing needs to happen asynchronously, for example, when triggering a user notification, generating reports, or processing a payment order. With this endpoint, we provide an example of how users can implement trace-based testing for such scenarios.

> Here the process in split in two phases: 

Reviewer: ```suggestion
Here the process is split into two phases: 
```

> 1. an API call that enqueue a import request to a queue

Reviewer: ```suggestion
1. An API call that enqueues an import request to a queue.
```

sequenceDiagram
    participant Endpoint as POST /pokemon
    participant API as API
    participant Queue as RabbitMQ
    
    Endpoint->>API: request

    alt request is invalid
        API-->>Endpoint: 400 Bad Request <br> <List of errors>
    end

    API->>Queue: enqueue "import" message
    Queue-->>API: message queued

    API-->>Endpoint: 201 Created
```

2. a Worker that dequeue messages and do the async process
```mermaid
sequenceDiagram
    participant Queue as RabbitMQ
    participant Worker as Worker
    participant ExternalAPI as PokeAPI
    participant Database as Postgres
    
    Queue->>Worker: dequeue "import" message

    Worker->>ExternalAPI: get pokemon info
    ExternalAPI-->>Worker: pokemon info

    Worker->>Database: save pokemon
    Database-->>Worker: pokemon saved
  
    alt if successful
      Worker-->>Queue: ack <br> queue can delete message
    else is failed
      Worker-->>Queue: nack <br> queue keep message for retries
    end
```

You can trigger this use case by calling the endpoint `POST /pokemon/import`, with the following request body:
```json
{
  "id":  52
}
```

It should return the following payload:
```json
{
  "id":  52
}
```

> ## Building a test for this scenario

Reviewer: ```suggestion
## Building a Test for This Scenario
```

Using Tracetest, we can [create a test](../../../web-ui/creating-tests.md) that will execute an API call on `POST /pokemon/import` and validate the following properties:

> - The API should enqueue an import task and return HTTP 200 OK

Reviewer: ```suggestion
- The API should enqueue an import task and return HTTP 200 OK.
```

> - The worker should dequeue the import task

Reviewer: ```suggestion
- The worker should dequeue the import task.
```

> - PokeAPI should return a valid response

Reviewer: ```suggestion
- PokeAPI should return a valid response.
```

- The database should respond with low latency (< 200ms)

### Traces

Running these tests for the first time will create an Observability trace like the image above, where you can see spans for the API call, the queue messaging, the PokeAPI (external API) call, and database calls. One interesting thing about this trace is that **you can observe the entire use case, end to end**:

![](../images/import-pokemon-trace.png)

### Assertions

> With this trace, now we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate the API and Worker behaviors:

Reviewer: ```suggestion
With this trace, we can build [assertions](../../../concepts/assertions.md) on Tracetest and validate the API and Worker behaviors:
```
